File: 10141 Request for Proposal/solution.c
#include <stdio.h>
#include <string.h>

int main() {
    int RFP = 0, n, p;
    char buffer[82], name[82];
    while(scanf("%d %d ", &n, &p) && n != 0 && p != 0) {
        if(RFP)
            putchar('\n');
        while(n--)
            fgets(buffer, sizeof(buffer), stdin);
        double minR;
        int maxD = -1;
        while(p--) {
            fgets(buffer, sizeof(buffer), stdin);
            double r;
            int d;
            scanf("%lf %d ", &r, &d);
            if(d > maxD) {
                strcpy(name, buffer);
                minR = r;
                maxD = d;
            } else if(d == maxD && r < minR) {
                strcpy(name, buffer);
                minR = r;
            }
            while(d--)
                fgets(buffer, sizeof(buffer), stdin);
        }
        printf("RFP #%d\n%s", ++RFP, name);
    }
    return 0;
}

File: 10141 Request for Proposal/solution.java
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
		int RFP = 0;
		String name = "";
		StringBuilder output = new StringBuilder();
		while(st.nextToken() == StreamTokenizer.TT_NUMBER) {
		    int n = (int)st.nval;
		    st.nextToken();
		    int p = (int)st.nval;
		    if(n == 0 && p == 0)
		        break;
		    if(RFP > 0)
		        output.append('\n');
		    while(n-- > 0)
		        br.readLine();
		    double minR = 0;
		    int maxD = -1;
		    while(p-- > 0) {
		        String s = br.readLine();
		        st.nextToken();
		        double r = st.nval;
		        st.nextToken();
		        int d = (int)st.nval;
		        if(d > maxD) {
		            name = new String(s);
		            minR = r;
		            maxD = d;
		        } else if(d == maxD && r < minR) {
		            name = new String(s);
		            minR = r;
		        }
		        while(d-- > 0)
		            br.readLine();
		    }
		    output.append("RFP #").append(++RFP).append('\n').append(name).append('\n');
		}
		System.out.print(output);
	}
}
